src/components/AssetsTable/index.jsx

```diff
@@ -88,6 +95,33 @@ export class AssetsTable extends React.Component {
     });
   }
 
+  onLockClick(asset) {
+    this.props.toggleLockAsset(asset, this.props.courseDetails);
+  }
+
+  getImageThumbnail(thumbnail) {
+    const baseUrl = this.props.courseDetails.base_url ? this.props.courseDetails.base_url : '';
+    return thumbnail ? (<img src={`${baseUrl}${thumbnail}`} alt="Description not available" />) : 'Preview not available';
```

Comment:
It feels slightly weird to try and get an image thumbnail for an `undefined` object. Especially since the value returned if `thumbnail` is `undefined`is not an image per se, but a `string` value.

Is there any reason why this control logic can't live a level above like

```javascript
if (currentAsset.thumbnail) { 
  currentAsset.image_preview = this.getImageThumbnail(currentAsset.thumbnail) } 
else { 
  currentAsset.image_preview = 'Preview not available' 
}
```

Reply:
I'm going to punt on this one for now. The only reason this code was marked as new is because I moved some functions around, and there's a bunch of other decision I want to tackle in this PR. Thanks for the suggestion though!